unmarshal.go

```diff
@@ -137,5 +156,28 @@ func readElement(r0 io.Reader, n int64, vo reflect.Value) (io.Reader, error) {
 				}
 			}
 		}
+		pos += headerSize + size
+	}
+}
+
+type UnmarshalOption func(*UnmarshalOptions)
+
+type UnmarshalOptions struct {
+	hooks []func(elem *Element)
+}
+
+// Element represents an EBML element
+type Element struct {
+	Value    interface{}
+	Name     string
+	Position uint64
+	Size     uint64
+	Parent   *Element
+}
+
+// WithElementHooks creates an UnmarshalOptions in which element hooks are registered
+func WithElementHooks(hooks ...func(*Element)) UnmarshalOption {
```

Comment:
`WithElementHooks` may make naming collision with options for `Marshal`.
(I think it would be also useful to have a hook in marshaller.)

How about `WithElementReadHooks` or `WithUnmarshalElementHooks`? (although it's bit long.)

Reply:
`WithElementReadHooks` sounds good to me, I will rename it 😃

Reply:
Renamed by 48ad2c779397426d77aba663be2a0fc0b3872f7e